akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java

```diff
@@ -240,22 +239,50 @@ public boolean defined(Object o) {
     return this;
   }
 
+  /**
+   * Add a new case statement to this builder.
+   *
+   * @param object the object to compare equals with
+   * @param externalPredicate an external predicate that will be evaluated if the object compares
+   *     equal
+   * @param apply an action to apply to the argument if the object compares equal
+   * @return a builder with the case statement added
+   */
+  @SuppressWarnings("unchecked")
+  public <P> ReceiveBuilder matchEquals(
+      final P object,
+      final java.util.function.BooleanSupplier externalPredicate,
```

Comment:
I don't see much value of having the overload, but we have `BooleanSupplier` in for example `match` so I guess it's good to add here also to make it consistent.

The syntactic difference is `notUsed -> !nodes.isEmpty()` vs `() -> !nodes.isEmpty()`.